compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -653,8 +652,23 @@ class ExpressionCodegen(
                 condition = negationArgument(condition as IrCall)
                 jumpIfFalse = false
             }
-            gen(condition, data).put(condition.asmType, mv)
-            BranchedValue.condJump(StackValue.onStack(condition.asmType), elseLabel, jumpIfFalse, mv)
+            if (isNullCheck(condition)) {
```

Comment:
Maybe extract whole if (!elseBranch)  branch body to separate method: 'genConditionWithOptimizationsIfPossible()'

Reply:
Done, thanks.